table.csv

```diff
@@ -21,7 +21,7 @@ keccak-512,                     multihash,      0x1d,           draft,
 blake3,                         multihash,      0x1e,           draft,     BLAKE3 has a default 32 byte output length. The maximum length is (2^64)-1 bytes.
 sha2-384,                       multihash,      0x20,           permanent, aka SHA-384; as specified by FIPS 180-4.
 dccp,                           multiaddr,      0x21,           draft,
-murmur3-128,                    multihash,      0x22,           draft,
+murmur3-x64-64,                 multihash,      0x22,           permanent, The first 64-bits of a murmur3-x64-128 - used for UnixFS directory sharding.
```

Comment:
_(I'll also note that https://github.com/aappleby/smhasher/issues/6 has the erroneous output for Guava as `4cae51b5316602c01c7c5642843e5fe7`, which doesn't figure in any of the above, it seems to be an actual error in that implementation)._

Reply:
> But is that something we want to document here?

@rvagg my understanding is that generally the that a major point of the multicodec table is to be able to have different groups reuse and implement the same schemes and identify their data the same way such that they can build compatible systems. In the case of multihash I'd generally assume that we're specifying data of the form `<hash code><hash len><hash bytes>` and how a user could take in a set of bytes and compute the same multihash output.

As a result, it seems like knowing how to order the output bytes to create the codec seems important for reproducibility. Am I missing something?

In general I'd hope that we could have some criteria for most codec entries which is to be able to point people at a spec (even if it's a living one) for how to implement these things and what the human meaning behind any particular 0xABCD actually means.

--- 

My main objections to what was merged here were:
1. Labeling 0x1022 as having a code despite it being both unspecified and unrequested
   - It doesn't personally matter to me what we specify, but I'd hope the next unfortunate soul to come along here and try to implement this could figure out whether the spaolacci approach (i.e. big endian) or the https://github.com/phensley/python-smhasher (i.e. small endian) was the one they should use to produce compatible results
   - We should be trying to make things better over time not worse, so let's try and push towards specifying what is meant by a new code.
2. Labeling 0x22 as murmur3-x64-64 despite knowing that this was incompletely specified and could still lead to confusion.
   - So let's label what we're doing. We can throw it in the notes field if we don't want to add more to the label name (and let any future soul who wants murmur3-x64-64-little-endian claim that if they want it).